src/signTxUtils.c

```diff
@@ -3,10 +3,40 @@
 #include "uiHelpers.h"
 #include "utils.h"
 #include "signTxUtils.h"
+#include "securityPolicy.h"
+#include "state.h"
 
 void respondSuccessEmptyMsg()
 {
 	TRACE();
 	io_send_buf(SUCCESS, NULL, 0);
 	ui_displayBusy(); // needs to happen after I/O
 }
+
+bool violatesSingleAccountOrStoreIt(const bip44_path_t* path)
+{
+	single_account_data_t* singleAccountData = &(instructionState.signTxContext.commonTxData.singleAccountData);
+
+	if (!bip44_hasOrdinaryWalletKeyPrefix(path) || !bip44_containsAccount(path)) {
+		TRACE("Invalid path in single account check");
+		ASSERT(false);
+	}
+	const bool byron = bip44_hasByronPrefix(path);
+	const uint32_t account = bip44_getAccount(path);
+	if (singleAccountData->pathSet) {
+		const uint32_t storedAccount = singleAccountData->accountNumber;
+		if (account != storedAccount) {
+			return true;
+		}
+		const bool combinesByronAndShelley = singleAccountData->byron ^ byron;
```

Comment:
Wouldn't `singleAccountData->byron != byron;` be a more standard way of checking unequality? I am not mistaken the result would be the same?

Reply:
I feel that the xor operator expresses the intent more clearly: either this or that is true but not both. Inequality is better when there are more than two options for values, no?

Reply:
Well, xor evokes some vague fear of hard-to-understand bitwise magic in me :D 
So I find inequality easier and faster to comprehend.

Reply:
Easy fix.